### PR TITLE
Artifactory trigger, Setup / External Configuration updates

### PR DESCRIPTION
**Pull request by fifthposition**
For the Artifactory trigger, clarify that it only works with Maven artifacts for now.

For external configuration, update links, add notes, and add info on using an S3 backend with Config Server.

cc @scottfrederick

### CODE REVIEW

**Comment by scottfrederick** [comment]
Note that the external configuration changes will be available in Spinnaker 1.18 (code PR pending).

File: setup/configuration/index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ sidebar:
...
+External configuration uses the open-source [Spring Cloud Config](https://spring.io/projects/spring-cloud-config) project, which includes a configuration server and client libraries to support centralized external configuration in a distributed system. The Config Server connects to a remote configuration repository and serves stored configuration properties to client applications. Configuration properties can be organized by Spring profile and application name and can be stored in any of a number of backends, including Git, HashiCorp Vault, AWS S3, JDBC, CredHub, and others. For more information about Spring Cloud Config, see the [documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html).
```
Comment (dorbin) [inline setup/configuration/index.md]:
```suggestion
External configuration uses the open-source [Spring Cloud Config](https://spring.io/projects/spring-cloud-config) project, which includes a configuration server and client libraries to support centralized external configuration in a distributed system. The Config Server connects to a remote configuration repository and serves stored configuration properties to client applications. 

Configuration properties can be organized by Spring profile and application name and can be stored in any of a number of backends, including GitHub, HashiCorp Vault, AWS S3, JDBC, CredHub, and others. For more information about Spring Cloud Config, see the [documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html).
```

File: setup/configuration/index.md [hunk 2]
```diff
@@ -62,7 +66,33 @@ spring:
           token: [vault access token]
 ```
 
-For information about configuring other supported Config Server backends, see the [Spring Cloud Config Server documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.1.0.RELEASE/single/spring-cloud-config.html#_environment_repository).
+> The above example configures authentication for Vault using the Vault
+> `X-Config-Token` header, via the setting of the
+> `spring.cloud.config.server.vault.token` property. In Spinnaker 1.18, you  can
```
Comment (dorbin) [inline setup/configuration/index.md]:
```suggestion
> `spring.cloud.config.server.vault.token` property. Starting with Spinnaker 1.18, you  can
```

File: setup/configuration/index.md [hunk 3]
```diff
@@ -62,7 +66,33 @@ spring:
...
+    config:
+      server:
+        awss3:
+          region: us-west-2
+          bucket: my-bucket
+```
+
+To configure authentication, you must use AWS's Default Credential Provider Chain. See the [AWS SDK documentation](https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html) for more information.
+
+For information about configuring other supported Config Server backends, see the [Spring Cloud Config Server documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html#_environment_repository).
```
Comment (dorbin) [inline setup/configuration/index.md]:
```suggestion
For information about configuring other supported config-server backends, see the [Spring Cloud Config Server documentation](https://cloud.spring.io/spring-cloud-static/spring-cloud-config/2.2.1.RELEASE/single/spring-cloud-config.html#_environment_repository).
```

File: setup/configuration/index.md [hunk 4]
```diff
@@ -160,6 +190,9 @@ ci:
 
 The Kubernetes, Google Cloud, and App Engine cloud providers can load account information from files that are separate from the service YAML configuration. You can load these separate files from an external source using the Config Server's Resource abstraction. In the service YAML configuration, prefix a file path with `configserver:` to indicate that the file should be retrieved by the Config Server.
```
Comment (dorbin) [inline setup/configuration/index.md]: There's no Google Cloud provider, per se. There's Compute Engine, App Engine (which is listed here), and Kubernetes Engine. I assume this is a reference to Compute Engine?

**Comment by dorbin** [comment]
Hi, @scottfrederick, can you drop in a link to that pending PR? Thanks!

File: setup/configuration/index.md (hunk 2 above)
Comment (scottfrederick) [reply setup/configuration/index.md]: replace
`using the Vault X-Config-Token header, via the setting ...`
with
`using a static authentication token, via the setting ...`

File: setup/configuration/index.md (hunk 3 above)
Comment (fifthposition) [reply setup/configuration/index.md]: I think I use "Config Server" generally as a shortened form of the project name (Spring Cloud Config Server), less as a generic term for something that serves configuration. Is that use OK?

File: setup/configuration/index.md (hunk 4 above)
Comment (fifthposition) [reply setup/configuration/index.md]: Fixed. Thanks!

**Review by dorbin — APPROVED** [review]
Thanks Ben!

**Comment by scottfrederick** [comment]
Corresponding code updates are in https://github.com/spinnaker/kork/pull/455

**Pull request merged**